src/ast/json.rs

```diff
@@ -80,24 +80,33 @@ impl<'arena> Ast<'arena> for JSON<'arena> {
     ) -> Vec<RecTok<'arena, Self>> {
         let is_pretty = format_style == &JSONFormat::Pretty;
         match self {
-            JSON::True => vec![RecTok::Tok(DisplayToken::Text("true".to_string()))],
-            JSON::False => vec![RecTok::Tok(DisplayToken::Text("false".to_string()))],
-            JSON::Null => vec![RecTok::Tok(DisplayToken::Text("null".to_string()))],
-            JSON::Str(string) => vec![RecTok::Tok(DisplayToken::Text(format!(r#""{}""#, string)))],
+            JSON::True => vec![RecTok::Tok(DisplayToken::Text("true".to_string(), "const"))],
+            JSON::False => vec![RecTok::Tok(DisplayToken::Text(
+                "false".to_string(),
```

Comment:
I believe `.to_owned()` is better than `.to_string()`

* Better conveys what's the point (`"..."` is already a string)
* Doesn't rely on specialization to be performant

However, I wonder if it's feasible to make it not allocate at all (by using `Cow<'static, str>`). IDK the specifics of `RecTok` or `DisplayToken` and don't have the time to study it right now.

Reply:
We aren't going to be modifying `DisplayToken`s after they're returned, so I'm thinking we have `DisplayToken::Text(String, SyntaxCategory)` and `DisplayToken::TextStr(&'static str, SyntaxCategory)`, which would stop us from having to spam `Cow::from` everywhere.

Reply:
Ok it's been `Cow`ified.